src/props.ts

```diff
@@ -0,0 +1,219 @@
+import cx from "classnames";
+import { HTMLProps } from "react";
+
+export function kebabCaseToCamelCase(str: string) {
+  return str.replaceAll(/-[a-z]/g, (g) => g[1]?.toUpperCase() ?? "");
+}
+
+/**
+ * Converts a CSS style string to an object
+ * that can be passed to a React component's
+ * `style` prop.
+ */
+export function cssToStyles(css: string) {
+  const cssJson = `{"${css
+    .replace(/;? *$/, "")
+    .replace(/;+ */g, '","')
+    .replace(/: */g, '":"')}"}`;
+
+  const obj = JSON.parse(cssJson);
+
+  return Object.keys(obj).reduce((acc, key) => {
+    const camelCased = key.startsWith("--") ? key : kebabCaseToCamelCase(key);
+    return { ...acc, [camelCased]: obj[key] };
+  }, {});
+}
+
+/**
+ * Merges two sets of React props. Class names
+ * will be concatenated and style objects
+ * will be merged.
+ */
+export function mergeReactProps(
+  a: HTMLProps<HTMLElement>,
+  b: HTMLProps<HTMLElement>
+) {
+  return {
+    ...a,
+    ...b,
+    className: cx(a.className, b.className),
```

Comment:
Is `cx` overkill here? Could we just do `${a.className} ${b.className}`?

Reply:
`className` could be `undefined`, so it would be like slightly more than that; feels just clunky enough to use `cx` for haha

Reply:
Fine for me.